canary/webrtc-c/src/CloudwatchMonitoring.cpp

```diff
@@ -233,26 +233,28 @@ VOID CloudwatchMonitoring::pushInboundRtpStats(Canary::PIncomingRTPMetricsContex
     this->push(incomingFrameDropRateDatum);
 }
 
-VOID CloudwatchMonitoring::pushEndToEndMetrics(Canary::PEndToEndMetricsContext pEndToEndMetricsContext)
+VOID CloudwatchMonitoring::pushEndToEndMetrics(Canary::EndToEndMetricsContext ctx)
 {
     MetricDatum endToEndLatencyDatum, sizeMatchDatum;
-
+    DOUBLE latency = ctx.frameLatencyAvg / (DOUBLE) HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
+
+    // TODO: due to https://github.com/aws-samples/amazon-kinesis-video-streams-demos/issues/96,
+    //       it's not clear why the emitted metric shows -nan. Since -nan is a string that's outputted
+    //       from Datum's value stringify implementation, we should try to get the original value by
+    //       printing it ourself.
+    //
+    //       If the issues doesn't exist anymore, please remove this as this is intended for debugging only.
+    //       The generic metric logging should be sufficient.
+    DLOGD("Current end-to-end frame latency: %4.2lf", latency);
```

Comment:
This is logged in the common functionality right? We will still be logging the aggregate here.

Reply:
It's as what the comment says, the common logging uses C++ string, which is entirely dependent on the implementor. What we want is the original value.

Reply:
Ah! Ok. Now I get it.